packages/react/src/components/cookieConsent/cookieConsentController.ts

```diff
@@ -0,0 +1,259 @@
+import _pick from 'lodash.pick';
+import _isObject from 'lodash.isobject';
+import _isUndefined from 'lodash.isundefined';
+
+import cookieControllerModule, { CookieSetOptions } from './cookieController';
+
+export type ConsentList = string[];
+
+export type ConsentObject = { [x: string]: boolean };
+
+export type ConsentStorage = {
+  required: ConsentObject;
+  optional: ConsentObject;
+  unknown?: ConsentObject;
+};
+
+export type ConsentControllerProps = {
+  requiredConsents?: ConsentList;
+  optionalConsents?: ConsentList;
+  cookieDomain?: string;
+};
+
+export type ConsentController = {
+  getRequired: () => ConsentObject;
+  getOptional: () => ConsentObject;
+  update: (key: string, value: boolean) => void;
+  approveRequired: () => void;
+  approveAll: () => void;
+  rejectAll: () => void;
+  getUnhandledConsents: () => string[];
+  getRequiredWithoutConsent: () => string[];
+  save: () => ConsentObject;
+};
+
+export const COOKIE_NAME = 'city-of-helsinki-cookie-consents';
+export const COOKIE_EXPIRATION_TIME = 60 * 60 * 24 * 365;
+
+export const commonConsents = {
+  matomo: 'matomo',
+  tunnistamo: 'tunnistamo',
+  language: 'language',
+  marketing: 'marketing',
+  preferences: 'preferences',
+};
+
+function convertStringArrayToKeyConsentObject(array: string[]): ConsentObject {
+  return array.reduce((current, key) => {
+    // eslint-disable-next-line no-param-reassign
+    current[key] = false;
+    return current;
+  }, {} as ConsentObject);
+}
+
+function mergeConsents(set1: ConsentObject, set2: ConsentObject, set3?: ConsentObject): ConsentObject {
+  return { ...set1, ...set2, ...set3 };
+}
+
+function parseConsents(jsonString: string | undefined): ConsentObject {
+  if (!jsonString || jsonString.length < 2 || jsonString.charAt(0) !== '{') {
+    return {};
+  }
+  try {
+    return JSON.parse(jsonString);
+  } catch (e) {
+    return {};
+  }
+}
+
+function createConsentsString(consents: ConsentObject): string {
+  if (!_isObject(consents)) {
+    return '{}';
+  }
+  return JSON.stringify(consents);
+}
+
+function createCookieController(
```

Comment:
Should we move this function inside the file cookieController.ts? It sounds like a function that user would expect to find there. If we move it there, we would not have to deal with problems that arise from duplicate name cookieController. Im referring to lines 96 and 99, where we have had to rename cookieController to `const get = (): string => cookieControllerModule.get(COOKIE_NAME) || '';`

Reply:
cookieController originally replaced universal-cookie and had only its methods. 

Good idea to add the creation there.